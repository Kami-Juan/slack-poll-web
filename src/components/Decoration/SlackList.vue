<template>
  <ul id="options-list">
    <slot />
  </ul>
</template>
<script>
export default {}
</script>
<style scoped>
#options-list {
  text-align: left;
  list-style-type: none;
  margin: 0;
  padding: 0;
}
</style>