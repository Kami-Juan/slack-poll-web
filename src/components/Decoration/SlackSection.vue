<template>
  <div class="slack-section">
    <div class="separator"></div>
    <div class="slack-slot">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SlackSection'
}
</script>

<style>
.slack-section {
  text-align: left;
  display: flex;
}

.separator {
  width: 4px;
  align-items: stretch;
  background-color: var(--yellowme-color);
  border-radius: 2px;
}

.slack-slot {
  padding-top: 3px;
  padding-bottom: 4px;
  margin-left: 13px;
}
</style>