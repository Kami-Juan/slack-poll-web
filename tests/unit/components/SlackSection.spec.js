import { shallowMount } from '@vue/test-utils'
import SlackList from './../../../src/components/Decoration/SlackList'
import SlackSection from './../../../src/components/Decoration/SlackSection'

describe('test SlackSection.vue', () => {
  let cmp

  beforeEach(() => {
    const slackListWrapper = {
      render(h) {
        return h(SlackList, { props: { value: 'One element' } })
      }
    }

    cmp = shallowMount(SlackSection, {
      slots: {
        default: slackListWrapper
      }
    })
  })

  it('should be render', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('SlackList are inserted in the component', () => {
    const slackSection = cmp.find(SlackSection)
    expect(slackSection.find(SlackList).isVueInstance()).toBe(true)
  })
})