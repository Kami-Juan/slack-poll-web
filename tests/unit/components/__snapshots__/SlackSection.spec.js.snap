// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SlackSection.vue should be render 1`] = `
<div
  class="slack-section"
>
  <div
    class="separator"
  />
   
  <div
    class="slack-slot"
  >
    <anonymous-stub />
  </div>
</div>
`;
